### PR TITLE
NOISSUE - Add group service tests

### PR DESCRIPTION
**Pull request by rodneyosodo**
# What type of PR is this?

This is a feature PR because it adds service tests to the `groups` package

## What does this do?

Adds service tests to the `groups` package

## Which issue(s) does this PR fix/relate to?

- Related Issue #
- Resolves #

## Have you included tests for your changes?

Yes, I have included tests for my changes.

## Did you document any new/modified feature?

No

### Notes

To be merged after https://github.com/absmach/magistrala/pull/197

### CODE REVIEW

File: internal/groups/service.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func (svc service) CreateGroup(ctx context.Context, token, kind string, g groups
 		})
 	}
 	if _, err := svc.auth.AddPolicies(ctx, &policies); err != nil {
-		return groups.Group{}, err
+		return g, err
```
Comment (arvindh123) [inline internal/groups/service.go:107]: Could you please explain here why we want to return varaible on error?

Comment (rodneyosodo) [reply internal/groups/service.go:107]: This is because we have already created the group and we don't have a rollback if AddPolicies fails.

Comment (arvindh123) [reply internal/groups/service.go:107]: Actually, Here creation flow need to corrected as per this ticket https://github.com/absmach/magistrala/issues/58

**Review by arvindh123 — APPROVED** [review]

**Review by arvindh123 — APPROVED** [review]

**Review by arvindh123 — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**